wfexecutor/__init__.py

```diff
@@ -402,22 +402,28 @@ def process_allowed_errors(allowed_errors_dict, wf_from_json):
 
     return allowed_errors_state
 
-def produce_versions_file(gi, workflow_from_json, path):
+def produce_versions_file(gi, workflow_from_json, path, tools_dict):
```

Comment:
```suggestion
def produce_versions_file(gi, workflow_from_json, path, tools_dict=[]):
```
Even better, just an empty default.

Reply:
but then wouldn't it be overwritten when entering a recursion @pcm32 ?